dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/EPersonPasswordAddOperation.java

```diff
@@ -66,7 +71,13 @@ public R perform(Context context, R object, Operation operation) {
             if (StringUtils.isNotBlank(token)) {
                 verifyAndDeleteToken(context, eperson, token, operation);
             }
-            ePersonService.setPassword(eperson, (String) operation.getValue());
+
+            String newPassword = (String) operation.getValue();
+            if (!validatePasswordService.isPasswordValid(context, newPassword)) {
+                throw new PasswordNotValidException("The new password to set is not valid");
```

Comment:
Since this exception's message has moved to `DSpaceApiExceptionController`, we could consider changing this to simply `throw new PasswordNotValidException()`.   This same feedback applies to everywhere this exception is thrown. We don't need a message for this exception unless it provides more detailed information.